### PR TITLE
Action: Add 13.2.Rel1 arm-none-eabi-gcc release.

### PR DESCRIPTION
**Pull request by tannewt**

### CODE REVIEW

**Comment by carlosperate** [comment]
Wow, that was really quick, thanks @tannewt!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -133,6 +134,6 @@ jobs:
     - name: Print out version
       run: arm-none-eabi-gcc --version
     - name: Check the version matches
-      run: arm-none-eabi-gcc --version | grep -q 12.3.Rel1
+      run: arm-none-eabi-gcc --version | grep -q 13.2.Rel1
```
Comment (carlosperate) [inline .github/workflows/test.yml:137]: Looks like the output --version in this release hasn't capitalised `Rel`:
```
$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (Arm GNU Toolchain 13.2.rel1 (Build arm-13.7)) 13.2.1 20231009
...

We can probably add the `-i` flag to grep to get this to pass.
```suggestion
      run: arm-none-eabi-gcc --version | grep -qi 13.2.Rel1
```
```

**Comment by tannewt** [comment]
> Wow, that was really quick, thanks @tannewt!

I'm hoping it fixes my CircuitPython code size issues. :-) My local machine has been on 13.2 for a while and it looks better there.

File: .github/workflows/test.yml (hunk 1 above)
Comment (carlosperate) [inline .github/workflows/test.yml:137]: I think this line is still missing the `i` flag.

**Comment by tannewt** [comment]
Ok, I think I got the right spot. Want me to rebase this together?

**Comment by carlosperate** [comment]
No worries, I can "squash and merge" the PR when the tests finish (I think they should all pass now) 👍

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -133,6 +134,6 @@ jobs:
     - name: Print out version
       run: arm-none-eabi-gcc --version
     - name: Check the version matches
-      run: arm-none-eabi-gcc --version | grep -q 12.3.Rel1
+      run: arm-none-eabi-gcc --version | grep -qi 13.2.Rel1
     - name: Check output path is correct
-      run: echo "${{ steps.action-under-test-id.outputs.path }}" | grep -q /home/runner/gcc-arm-none-eabi-12.3.1-linux/arm-gnu-toolchain-12.3.rel1-x86_64-arm-none-eabi/bin
+      run: echo "${{ steps.action-under-test-id.outputs.path }}" | grep -q /home/runner/gcc-arm-none-eabi-13.2.1-linux/arm-gnu-toolchain-13.2.rel1-x86_64-arm-none-eabi/bin
```
Comment (carlosperate) [inline .github/workflows/test.yml:139]: A bit of a pain the the releases are never that consisten, but I think this should do it:
```suggestion
      run: echo "${{ steps.action-under-test-id.outputs.path }}" | grep -q /home/runner/gcc-arm-none-eabi-13.2.1-linux/arm-gnu-toolchain-13.2.Rel1-x86_64-arm-none-eabi/bin
```

Comment (carlosperate) [reply .github/workflows/test.yml:139]: This is the last one and we have to squash anyway, so I'll merge this commit to the PR from here and merge the PR when we finally get the all-green.

**Comment by carlosperate** [comment]
All tests pass 🎉 
Thank you for the PR @tannewt, this is great!
I'll merge and release right away 👍

**Pull request merged**